src/sentry/workflow_engine/processors/workflow.py

```diff
@@ -69,19 +77,54 @@ def evaluate_workflow_triggers(workflows: set[Workflow], job: WorkflowJob) -> se
     workflows_to_enqueue: set[Workflow] = set()
 
     for workflow in workflows:
-        if workflow.evaluate_trigger_conditions(job):
-            triggered_workflows.add(workflow)
+        evaluation, remaining_conditions = workflow.evaluate_trigger_conditions(job)
+        if remaining_conditions:
+            workflows_to_enqueue.add(workflow)
```

Comment:
i think we should enqueue the same thing in the buffer as in `evaluate_workflow_action` filters, but i assume this is happening in a follow up PR

Reply:
Yep! rather than us trying to create a data structure here and then enqueue a bunch at the end, i ended up just enqueuing directly in the other PR. that made it so we don't need to have an additional query to match the workflows to the data condition groups when enqueuing them.